### ISSUE TITLE
Docker installation script now tells users to install Docker Desktop for Windows when run in WSL

### ISSUE THREAD

**Issue by adrastogi** [MicrosoftDocs/WSL#1705]
### Documentation Issue

Noticed this when setting up a new machine a little while back... the script at https://get.docker.com now explicitly checks if it's being invoked from WSL, and if so, instructs the user to install Docker Desktop for Windows instead:

From the script:
```
        if is_wsl; then
                echo
                echo "WSL DETECTED: We recommend using Docker Desktop for Windows."
                echo "Please get Docker Desktop from https://www.docker.com/products/docker-desktop"
                echo
                cat >&2 <<-'EOF'

                        You may press Ctrl+C now to abort this script.
                EOF
                ( set -x; sleep 20 )
        fi
```

### Link to documentation page

https://learn.microsoft.com/en-us/windows/wsl/tutorials/gpu-compute

### Suggested Improvements

I think step 2 of the instructions can be updated to just recommend installing Docker Desktop for Windows. (I am less confident about what additional changes are needed downstream of that, which is why I hesitated on sending a PR- for example, is it still necessary to install the NVIDIA Container Toolkit? If so, are the current instructions accurate? The NVIDIA document that is linked to doesn't seem to have the associated chapter anymore. Hopefully someone familiar with the updated recommendations can review the rest of the document for accuracy.)

**Comment by TheOnlyWei** [comment]
> Hey! So can't install anymore at all or just with this script?

Yes, the script doesn't install docker. I tried the script inside an Ubuntu container and it complained that I had WSL installed.

**Comment by linghengqian** [comment]
- I assume there are Docker Committers out there who can help review https://github.com/docker/docker-install/pull/475 ?
- The script never failed, just had a 20 second delay.
